IEP-729: ESP32-C2 and ESP32-H2 support in IDEs

sigmaaa opened the pull request:
## Description

Added esp32-c2 and esp32-h2 targets, which are using the same riscv toolchain as the esp32-c3 target, so it's safe to add these targets with older version esp-idf as well. 
Also, the bug fix related to changing the toolchain to the xtensa-clang was added to this PR

New feature # ([IEP-756](https://jira.espressif.com:8443/browse/IEP-756))

## Type of change
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How has this been tested?

Test 1:
- install tools from IDE
- check existing toolchains. `Eclipse -> Preferences -> C/C++ -> Core Build Toolchains`. There should be all toolchains including esp32-c2 and esp32-h2. Xtensa-clang toolchain has a clang type 
- select esp32-c2 or esp32-h2 and build a project with it. Check if a building is ok.

Test 2:
- select esp32 target
- edit the launch configuration. In the build tab select the xtensa-clang toolchain.
- build the project. Check if it's built with xtensa-clang toolchain

Test 3:
- Create a debug configuration
- edit the debug configuration. In the debugger tab, select esp32-c2 board (esp32-h2 board not yet in the list provided by openocd). 
- Check if the gdb executable is correct

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):


## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

kolipakakondal commented on bundles/com.espressif.idf.core/src/com/espressif/idf/core/build/ESP32CMakeToolChainProvider.java:
```diff
@@ -19,7 +19,7 @@ public class ESP32CMakeToolChainProvider extends AbstractESPCMakeToolChainProvid
 	{
-		return tcManager.getToolChain(ESPToolChainProvider.ID, ESP32ToolChain.ID);
+		return tcManager.getToolChain(ESPToolChainProvider.ID, ESP32ClangToolChain.ID);
 	}
```
Maybe this will return esp32clang toolchain for esp32 target? I think which is wrong!

sigmaaa replied:
fixed

kolipakakondal commented on bundles/com.espressif.idf.core/src/com/espressif/idf/core/build/ESP32ClangCmakeToolChainProvider.java:
```diff
@@ -0,0 +1,17 @@
+	protected IToolChain getToolchain() throws CoreException
+	{
+		return tcManager.getToolChain(ESPToolChainProvider.ID, ESP32ToolChain.ID);
+	}
```
Here we need to keep ESP32ClangToolChain.ID?

sigmaaa replied:
fixed

kolipakakondal commented on bundles/com.espressif.idf.core/src/com/espressif/idf/core/build/ESP32ClangToolChain.java:
```diff
@@ -0,0 +1,29 @@
+public class ESP32ClangToolChain extends ClangToolChain
+{
+
+	public static final String ID = "xtensa-esp32-elf"; //$NON-NLS-1$
```
xtensa-esp32-elf-clang ?

sigmaaa replied:
changed to `xtensa`

kolipakakondal reviewed (approved):
LGTM. Later I think we need to redesign these classes in a such way that based on the properties of each target we could return the ToolChain object dynamically instead of managing separate classes for each target. Otherwise, it will be difficult to manage over a period of time.

AndriiFilippov commented:
@sigmaaa hi ! 

Tested under: 
OS - Windows 10
ESP-IDF: master 

- Able to build project for esp32-c2 and esp32-h2 targets. LGTM 
- In `Eclipse -> Preferences -> C/C++ -> Core Build Toolchains`  could not see xtensa-clang toolchain. Could not build project. 
Please, see image below. 

![image](https://user-images.githubusercontent.com/69584713/193027132-126a9e04-1c90-4200-b4c9-6f3e5368654f.png)

sigmaaa commented:
> @sigmaaa hi !
> 
> Tested under: OS - Windows 10 ESP-IDF: master
> 
> * Able to build project for esp32-c2 and esp32-h2 targets. LGTM
> * In `Eclipse -> Preferences -> C/C++ -> Core Build Toolchains`  could not see xtensa-clang toolchain - only clang-gcc. Could not build project.
>   Please, see image below.
> 
> ![image](https://user-images.githubusercontent.com/69584713/193027132-126a9e04-1c90-4200-b4c9-6f3e5368654f.png)

Hi @AndriiFilippov,

Thanks for the review. The problem was that `xtensa-esp32-elf-clang.exe -v` command could not be executed without specifying the `PATH` environment variable on Windows. Fixed in the latest commit. Also deleted gcc toolchain, that is coming from clang directory to not confuse the user with it. PTAL

AndriiFilippov commented:
@sigmaaa hi ! 

Tested under: 
OS - Ubuntu 20.04 
ESP-IDF: master

- Able to build project for esp32-c2 and esp32-h2 targets. LGTM
- Able to see in `Eclipse -> Preferences -> C/C++ -> Core Build Toolchains` xtensa-clang toolchain. But could not build project.

There is some compiler warnings and errors: 



```
[336/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/newlib/priv_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/spi_flash/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu17 -MD -MT esp-idf/newlib/CMakeFiles/__idf_newlib.dir/stdatomic.c.obj -MF esp-idf/newlib/CMakeFiles/__idf_newlib.dir/stdatomic.c.obj.d -o esp-idf/newlib/CMakeFiles/__idf_newlib.dir/stdatomic.c.obj   -c /home/osboxes/e/esp-idf/components/newlib/stdatomic.c
[337/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/priv_include -I/home/osboxes/e/esp-idf/components/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/private_include -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/esp_pm/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu17 -MD -MT esp-idf/freertos/CMakeFiles/__idf_freertos.dir/FreeRTOS-Kernel/tasks.c.obj -MF esp-idf/freertos/CMakeFiles/__idf_freertos.dir/FreeRTOS-Kernel/tasks.c.obj.d -o esp-idf/freertos/CMakeFiles/__idf_freertos.dir/FreeRTOS-Kernel/tasks.c.obj   -c /home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/tasks.c
[338/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/newlib/priv_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/spi_flash/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu17 -MD -MT esp-idf/newlib/CMakeFiles/__idf_newlib.dir/port/esp_time_impl.c.obj -MF esp-idf/newlib/CMakeFiles/__idf_newlib.dir/port/esp_time_impl.c.obj.d -o esp-idf/newlib/CMakeFiles/__idf_newlib.dir/port/esp_time_impl.c.obj   -c /home/osboxes/e/esp-idf/components/newlib/port/esp_time_impl.c
[339/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/esp_timer/include -I/home/osboxes/e/esp-idf/components/esp_timer/private_include -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/driver/include -I/home/osboxes/e/esp-idf/components/driver/deprecated -I/home/osboxes/e/esp-idf/components/driver/esp32/include -I/home/osboxes/e/esp-idf/components/esp_pm/include -I/home/osboxes/e/esp-idf/components/esp_ringbuf/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu17 -MD -MT esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/system_time.c.obj -MF esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/system_time.c.obj.d -o esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/system_time.c.obj   -c /home/osboxes/e/esp-idf/components/esp_timer/src/system_time.c
[340/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang++  -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/pthread/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu++20 -fno-exceptions -fno-rtti -MD -MT esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_exception_stubs.cpp.obj -MF esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_exception_stubs.cpp.obj.d -o esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_exception_stubs.cpp.obj -c /home/osboxes/e/esp-idf/components/cxx/cxx_exception_stubs.cpp
[341/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang++  -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/pthread/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu++20 -fno-exceptions -fno-rtti -MD -MT esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_guards.cpp.obj -MF esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_guards.cpp.obj.d -o esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_guards.cpp.obj -c /home/osboxes/e/esp-idf/components/cxx/cxx_guards.cpp
FAILED: esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_guards.cpp.obj 
ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang++  -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/pthread/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu++20 -fno-exceptions -fno-rtti -MD -MT esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_guards.cpp.obj -MF esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_guards.cpp.obj.d -o esp-idf/cxx/CMakeFiles/__idf_cxx.dir/cxx_guards.cpp.obj -c /home/osboxes/e/esp-idf/components/cxx/cxx_guards.cpp
/home/osboxes/e/esp-idf/components/cxx/cxx_guards.cpp:170:17: error: '__cxa_guard_release' is missing exception specification 'throw()'
extern "C" void __cxa_guard_release(__guard* pg)
                ^
                                                 throw()
/home/osboxes/e/esp-idf/components/cxx/cxx_guards.cpp:29:17: note: previous declaration is here
extern "C" void __cxa_guard_release(__guard* pg);
                ^
/home/osboxes/e/esp-idf/components/cxx/cxx_guards.cpp:192:17: error: '__cxa_guard_abort' is missing exception specification 'throw()'
extern "C" void __cxa_guard_abort(__guard* pg)
                ^
                                               throw()
/home/osboxes/e/esp-idf/components/cxx/cxx_guards.cpp:30:17: note: previous declaration is here
extern "C" void __cxa_guard_abort(__guard* pg);
                ^
2 errors generated.
[342/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -DMBEDTLS_CONFIG_FILE=\"mbedtls/esp_config.h\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/ulp/ulp_common/include -I/home/osboxes/e/esp-idf/components/ulp/ulp_common/include/esp32 -I/home/osboxes/e/esp-idf/components/driver/include -I/home/osboxes/e/esp-idf/components/driver/deprecated -I/home/osboxes/e/esp-idf/components/driver/esp32/include -I/home/osboxes/e/esp-idf/components/esp_pm/include -I/home/osboxes/e/esp-idf/components/esp_ringbuf/include -I/home/osboxes/e/esp-idf/components/esp_adc/include -I/home/osboxes/e/esp-idf/components/esp_adc/interface -I/home/osboxes/e/esp-idf/components/esp_adc/esp32/include -I/home/osboxes/e/esp-idf/components/esp_adc/deprecated/include -I/home/osboxes/e/esp-idf/components/efuse/include -I/home/osboxes/e/esp-idf/components/efuse/esp32/include -I/home/osboxes/e/esp-idf/components/esp_http_client/include -I/home/osboxes/e/esp-idf/components/esp_http_server/include -I/home/osboxes/e/esp-idf/components/http_parser -I/home/osboxes/e/esp-idf/components/bootloader_support/include -I/home/osboxes/e/esp-idf/components/bootloader_support/bootloader_flash/include -I/home/osboxes/e/esp-idf/components/nvs_flash/include -I/home/osboxes/e/esp-idf/components/spi_flash/include -I/home/osboxes/e/esp-idf/components/esp_wifi/include -I/home/osboxes/e/esp-idf/components/esp_event/include -I/home/osboxes/e/esp-idf/components/esp_phy/include -I/home/osboxes/e/esp-idf/components/esp_phy/esp32/include -I/home/osboxes/e/esp-idf/components/esp_netif/include -I/home/osboxes/e/esp-idf/components/app_update/include -I/home/osboxes/e/esp-idf/components/esp_app_format/include -I/home/osboxes/e/esp-idf/components/wpa_supplicant/include -I/home/osboxes/e/esp-idf/components/wpa_supplicant/port/include -I/home/osboxes/e/esp-idf/components/wpa_supplicant/esp_supplicant/include -I/home/osboxes/e/esp-idf/components/esp_serial_slave_link/include -I/home/osboxes/e/esp-idf/components/sdmmc/include -I/home/osboxes/e/esp-idf/components/esp-tls -I/home/osboxes/e/esp-idf/components/esp-tls/esp-tls-crypto -I/home/osboxes/e/esp-idf/components/mbedtls/port/include -I/home/osboxes/e/esp-idf/components/mbedtls/mbedtls/include -I/home/osboxes/e/esp-idf/components/mbedtls/mbedtls/library -I/home/osboxes/e/esp-idf/components/mbedtls/esp_crt_bundle/include -I/home/osboxes/e/esp-idf/components/esp_https_ota/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu17 -MD -MT esp-idf/esp_common/CMakeFiles/__idf_esp_common.dir/src/esp_err_to_name.c.obj -MF esp-idf/esp_common/CMakeFiles/__idf_esp_common.dir/src/esp_err_to_name.c.obj.d -o esp-idf/esp_common/CMakeFiles/__idf_esp_common.dir/src/esp_err_to_name.c.obj   -c /home/osboxes/e/esp-idf/components/esp_common/src/esp_err_to_name.c
[343/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/esp_timer/include -I/home/osboxes/e/esp-idf/components/esp_timer/private_include -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/driver/include -I/home/osboxes/e/esp-idf/components/driver/deprecated -I/home/osboxes/e/esp-idf/components/driver/esp32/include -I/home/osboxes/e/esp-idf/components/esp_pm/include -I/home/osboxes/e/esp-idf/components/esp_ringbuf/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu17 -MD -MT esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/ets_timer_legacy.c.obj -MF esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/ets_timer_legacy.c.obj.d -o esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/ets_timer_legacy.c.obj   -c /home/osboxes/e/esp-idf/components/esp_timer/src/ets_timer_legacy.c
[344/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/esp_timer/include -I/home/osboxes/e/esp-idf/components/esp_timer/private_include -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/driver/include -I/home/osboxes/e/esp-idf/components/driver/deprecated -I/home/osboxes/e/esp-idf/components/driver/esp32/include -I/home/osboxes/e/esp-idf/components/esp_pm/include -I/home/osboxes/e/esp-idf/components/esp_ringbuf/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu17 -MD -MT esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/esp_timer.c.obj -MF esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/esp_timer.c.obj.d -o esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/esp_timer.c.obj   -c /home/osboxes/e/esp-idf/components/esp_timer/src/esp_timer.c
[345/841] ccache /home/osboxes/.espressif/tools/xtensa-clang/14.0.0-38679f0333/xtensa-esp32-elf-clang/bin/clang -DESP_PLATFORM -DIDF_VER=\"v5.1-dev-1006-g02605f1a31\" -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -Iconfig -I/home/osboxes/e/esp-idf/components/esp_timer/include -I/home/osboxes/e/esp-idf/components/esp_timer/private_include -I/home/osboxes/e/esp-idf/components/newlib/platform_include -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include/freertos -I/home/osboxes/e/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/osboxes/e/esp-idf/components/freertos/esp_additions/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc -I/home/osboxes/e/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/. -I/home/osboxes/e/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/osboxes/e/esp-idf/components/heap/include -I/home/osboxes/e/esp-idf/components/log/include -I/home/osboxes/e/esp-idf/components/soc/include -I/home/osboxes/e/esp-idf/components/soc/esp32/. -I/home/osboxes/e/esp-idf/components/soc/esp32/include -I/home/osboxes/e/esp-idf/components/hal/esp32/include -I/home/osboxes/e/esp-idf/components/hal/include -I/home/osboxes/e/esp-idf/components/hal/platform_port/include -I/home/osboxes/e/esp-idf/components/esp_rom/include -I/home/osboxes/e/esp-idf/components/esp_rom/include/esp32 -I/home/osboxes/e/esp-idf/components/esp_rom/esp32 -I/home/osboxes/e/esp-idf/components/esp_common/include -I/home/osboxes/e/esp-idf/components/esp_system/include -I/home/osboxes/e/esp-idf/components/esp_system/port/soc -I/home/osboxes/e/esp-idf/components/esp_system/port/include/private -I/home/osboxes/e/esp-idf/components/xtensa/include -I/home/osboxes/e/esp-idf/components/xtensa/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/include -I/home/osboxes/e/esp-idf/components/lwip/include/apps -I/home/osboxes/e/esp-idf/components/lwip/include/apps/sntp -I/home/osboxes/e/esp-idf/components/lwip/lwip/src/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include -I/home/osboxes/e/esp-idf/components/lwip/port/esp32/include/arch -I/home/osboxes/e/esp-idf/components/driver/include -I/home/osboxes/e/esp-idf/components/driver/deprecated -I/home/osboxes/e/esp-idf/components/driver/esp32/include -I/home/osboxes/e/esp-idf/components/esp_pm/include -I/home/osboxes/e/esp-idf/components/esp_ringbuf/include --target=xtensa -mcpu=esp32 -ffreestanding    -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -Wno-documentation -Wno-typedef-redefinition -Wno-char-subscripts -Wno-format-security -Wno-tautological-overlap-compare -Wno-tautological-pointer-compare -Wno-pointer-bool-conversion -Wno-string-concatenation -Wno-section -Wno-unknown-attributes -Wno-atomic-alignment -fmacro-prefix-map=/home/osboxes/Desktop/Espressif-IDE/workspace/tt=. -fmacro-prefix-map=/home/osboxes/e/esp-idf=/IDF -fno-jump-tables -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu17 -MD -MT esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/esp_timer_impl_lac.c.obj -MF esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/esp_timer_impl_lac.c.obj.d -o esp-idf/esp_timer/CMakeFiles/__idf_esp_timer.dir/src/esp_timer_impl_lac.c.obj   -c /home/osboxes/e/esp-idf/components/esp_timer/src/esp_timer_impl_lac.c
ninja: build stopped: subcommand failed.
Build complete (2 errors, 5 warnings): /home/osboxes/Desktop/Espressif-IDE/workspace/tt/build
/home/osboxes/.espressif/python_env/idf5.1_py3.8_env/bin/python /home/osboxes/e/esp-idf/tools/idf_size.py /home/osboxes/Desktop/Espressif-IDE/workspace/tt/build/tt.map
/home/osboxes/.espressif/python_env/idf5.1_py3.8_env/bin/python /home/osboxes/e/esp-idf/components/partition_table/gen_esp32part.py /home/osboxes/Desktop/Espressif-IDE/workspace/tt/build/partition_table/partition-table.bin
Total time taken to build the project: 53,827 ms
```

AndriiFilippov commented:
@sigmaaa 

Tested under: 
OS - Windows 10 
ESP-IDF: v4.4 

As I see from `idf.py --list-targets` command the release supports the following targets: 

- esp32
- esp32s2
- esp32c3
- esp32s3

For esp32sc2 I got: `No toolchain file found`
For esp32h2 - project builds and run without any warnings, is it ok ?
